osu.Game/Screens/Play/HUD/ClicksPerSecond/ClicksPerSecondCalculator.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Timing;
+using osu.Game.Rulesets.UI;
+
+namespace osu.Game.Screens.Play.HUD.ClicksPerSecond
+{
+    public class ClicksPerSecondCalculator : Component
+    {
+        private readonly List<double> timestamps;
+
+        private InputListener? listener;
+
+        [Resolved]
+        private IGameplayClock? gameplayClock { get; set; }
+
+        [Resolved(canBeNull: true)]
+        private DrawableRuleset? drawableRuleset { get; set; }
+
+        public InputListener Listener
+        {
+            set
+            {
+                onResetRequested?.Invoke();
+                listener = value;
+            }
+        }
+
+        private event Action? onResetRequested;
+
+        private IClock? workingClock => drawableRuleset?.FrameStableClock;
+
+        private double baseRate;
+
+        private double rate
```

Comment:
Should not exist

Reply:
https://user-images.githubusercontent.com/39100084/186424451-b2d01316-3b2d-46fb-be9d-6bbcddd3912c.mp4

Without this, the value will return 0 on pause (due to `IGameplayClock.TrueGameplayRate` returning 0). Maybe I should move the rate logic stuff in a separate method?